src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1291,15 +1291,15 @@ impl Disambiguator {
                 "enum" => Kind(DefKind::Enum),
                 "trait" => Kind(DefKind::Trait),
                 "union" => Kind(DefKind::Union),
-                "module" | "mod" => Kind(DefKind::Mod),
+                "mod" => Kind(DefKind::Mod),
                 "const" | "constant" => Kind(DefKind::Const),
                 "static" => Kind(DefKind::Static),
-                "function" | "fn" | "method" => Kind(DefKind::Fn),
+                "fn" | "method" => Kind(DefKind::Fn),
                 "derive" => Kind(DefKind::Macro(MacroKind::Derive)),
                 "type" => NS(Namespace::TypeNS),
                 "value" => NS(Namespace::ValueNS),
                 "macro" => NS(Namespace::MacroNS),
-                "prim" | "primitive" => Primitive,
+                "primitive" => Primitive,
```

Comment:
Can you also add a help message if someone tries to use the old names? I'd implement it something like this:

- Add a new `ResolutionError::UnsupportedDisambiguator` variant
- Return the variant if someone tries to use the old names (and change the error type of the function to `ResolutionFailure`):

```suggestion
                "primitive" => Primitive,
                "module" | "function" | "prim" => return Err(ResolutionError::UnsupportedDisambiguator(prefix)),
```

- Change `fn resolution_failure` to give a helpful error, something like

```
warning: unresolved link to `prim@char`
 --> bad.rs:1:6
  |
1 | //! [prim@char]
  |      ^^^^^^^^ no item named `prim@char` in `bad`
  |      ^^^^^^^^ help: the `prim` disambiguator has been removed, try `primitive` instead: `primitive@char`
  |
  = note: the `prim` disambiguator has been removed
```

There are examples in that function, but feel free to ask for help if you need it.

Reply:
I can do this, but I’m a little surprised you’d suggest it: this is niche, previously-unstable functionality, and it wouldn’t *surprise* me if literally no one has used these aliases.

Reply:
The problem with that reasoning is that people aren't _using_ it as if it's unstable. `module@` does not seem particularly niche to me and neither does `function@`. Part of the reason I've been trying to stabilize intra-doc links is because people _are_ treating it as if it's stable, because docs.rs uses the nightly compiler.

This is part of a larger discussion I want to have about rustdoc and nightly: rustdoc should be enforcing feature gates. Right now people don't know what's stable and what's unstable because very few people document locally, and those that do mostly use nightly and think everything must be fine on stable because they haven't added `#![feature]`.

Reply:
I *meant* the disambiguation prefixes, which I think it’s fair to say are niche; I had forgotten that intra-doc links as a whole were only now becoming stable!

Reply:
> I had forgotten that intra-doc links as a whole were only now becoming stable!

Right, this is what I mean by rustdoc should enforce feature gates. See also https://github.com/rust-lang/rust/issues/63305.

> I meant the disambiguation prefixes, which I think it’s fair to say are niche

Well, either way I don't think we should remove functionality without pointing to how to fix this code. So I'd strongly prefer to add this help message.